dll/win32/shell32/folders/CFSFolder.cpp

```diff
@@ -106,6 +106,11 @@ static LPCWSTR ExtensionFromPidl(PCUIDLIST_RELATIVE pidl, LPWSTR Buf, UINT cchMa
     return pExtension;
 }
 
+static LPCWSTR ExtensionFromPidl(PCUIDLIST_RELATIVE pidl, LPWSTR Buf, UINT cchMax)
```

Comment:
Since this is C++, what about keeping the existing `ExtensionFromPidl` function, but making its `AllowFolder` parameter optional, with the default value being FALSE ?
```c
static LPCWSTR ExtensionFromPidl(PCUIDLIST_RELATIVE pidl, LPWSTR Buf, UINT cchMax, BOOL AllowFolder = FALSE)
```